### PR TITLE
Fix invalid diagnostic prop_add

### PR DESCRIPTION
**Pull request by ippachi**
Related: https://github.com/prabirshrestha/vim-lsp/pull/721#issuecomment-753279052
Similar problem occurred solargraph too.
And I fixed in solargraph. But rust still error. I think rust problem reason is #1076

### before
before-ruby

### after
after-ruby

### CODE REVIEW

**Comment by prabirshrestha** [comment]
can you also include tests for multibyte chars.

//cc @hrsh7th in case there is already a util in the vital lsp that include in range.

File: autoload/lsp/utils/range.vim [hunk 1]
```diff
@@ -70,3 +70,23 @@ function! lsp#utils#range#lsp_to_vim(bufnr, range) abort
...
+
+    let l:first_line_length = len(getbufline(a:bufnr, l:start_line, l:end_line)[0])
+    let l:last_line_length = len(getbufline(a:bufnr, l:start_line, l:end_line)[-1])
+
+    if l:first_line_length < l:start_col || l:last_line_length < l:end_col
+        return v:false
+    endif
+
+    return v:true
+endfunction
```
Comment (hrsh7th) [inline autoload/lsp/utils/range.vim:92]: I think we should add `range#_get_buffer_range` and `range#_contains`.

The `range#_contains` implemented in here (this PR stale now due to nvim_extmark issues)
https://github.com/prabirshrestha/vim-lsp/pull/1022/files#diff-9f982ec39267f1f987caa856add5947c949fb085afd6c4679f47917f21ed88daR32

**Comment by ippachi** [comment]
Added a test, although it may no longer be needed because there is a solution of hrsh7th.

**Comment by prabirshrestha** [comment]
@ippachi could you fix lint issues and contains. That way we can reuse the code for linkedEditingRange too.

**Comment by ippachi** [comment]
OK I'll try it!! 😄

**Comment by ippachi** [comment]
This fix does not seem to have fixed the problem correctly.

### First
My code is wrong, line 88 of range.vim `l:last_line_length < l:end_col`, it should have been `l:last_line_length <= l:end_col`. In the former case, highlighting the last character of a line will not work.

The reason why the screenshot looks like it works is because prop_add is not executed due to this mistake.

### Second
`E685: Internal error: text property below deleted line not found`.
This error is occured by https://github.com/vim/vim/blob/59ff64079b4fcf0393a7413e3242a8b58c2ceadb/src/memline.c#L3524. And this function seems to be executed by deleting a line.
So it doesn't seem to be an error in prop_add as you say here https://github.com/prabirshrestha/vim-lsp/pull/721#issuecomment-753279052.


I think we should fix #1076 first and close this PR because code is wrong.
@prabirshrestha

**Pull request closed without merging**